oauth/src/main/java/ch/cyberduck/core/oauth/OAuth2AuthorizationService.java

```diff
@@ -135,25 +135,30 @@ public OAuthTokens authorize(final Host bookmark, final LoginCallback prompt, fi
         if(log.isDebugEnabled()) {
             log.debug(String.format("Start new OAuth flow for %s with missing access token", bookmark));
         }
-        final TokenResponse response;
+
+        final IdTokenResponse response;
+        // Save access token, refresh token and id token
         switch(flowType) {
             case AuthorizationCode:
                 response = this.authorizeWithCode(bookmark, prompt);
-                break;
+                return credentials.withOauth(new OAuthTokens(
```

Comment:
It has been discussed before that these two return statement are identical and can be moved below the switch.